src/core/mac/mac.cpp

```diff
@@ -2050,8 +2175,87 @@ uint8_t Mac::GetTimeIeOffset(const Frame &aFrame)
 }
 #endif
 
+#if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
+void Mac::SetCslChannel(uint8_t aChannel)
+{
+    VerifyOrExit(GetCslChannel() != aChannel, OT_NOOP);
+
+    mSubMac.SetCslChannel(aChannel);
+
+    if (IsCslEnabled())
+    {
+        Get<Mle::Mle>().ScheduleChildUpdateRequest();
+    }
+exit:
+    return;
+}
+
+void Mac::SetCslPeriod(uint16_t aPeriod)
+{
+    mSubMac.SetCslPeriod(aPeriod);
+    IgnoreError(Get<Radio>().EnableCsl(GetCslPeriod(), &Get<Mle::Mle>().GetParent().GetExtAddress()));
```

Comment:
There might be chance that  a non-zero `aPeriod` is set before attached, that is, the parent might be invalid and extaddr might be meaningless which was said required for nordic platform.

Reply:
It makes sense. Would solve it in next PR for CSL on Nordic.